components/scream/src/physics/rrtmgp/CMakeLists.txt

```diff
@@ -1,50 +1,58 @@
-# Add RRTMGP source files. Right now, this builds the Fortran source for RRTMGP.
-# Once the C++ port of RRTMGP is complete we should switch over to that.
+# RRTMGP++ requires YAKL
+add_subdirectory(${SCREAM_BASE_DIR}/../../externals/YAKL ${CMAKE_CURRENT_BINARY_DIR}/YAKL_build)
+target_compile_options(yakl PUBLIC $<$<COMPILE_LANGUAGE:Fortran>:-fno-default-real-8 -fno-default-double-8>)
```

Comment:
I think it has to do with overloaded procedures defined in YAKL. The pool allocator F90 interface overloads `real4` and `real8` parameters:
```
interface gator_deallocate
   ...
    module procedure :: gator_deallocate_real4_1d
    module procedure :: gator_deallocate_real4_2d
    module procedure :: gator_deallocate_real4_3d
    module procedure :: gator_deallocate_real4_4d
    module procedure :: gator_deallocate_real4_5d
    module procedure :: gator_deallocate_real4_6d
    module procedure :: gator_deallocate_real4_7d

    module procedure :: gator_deallocate_real8_1d
    module procedure :: gator_deallocate_real8_2d
    module procedure :: gator_deallocate_real8_3d
    module procedure :: gator_deallocate_real8_4d
    module procedure :: gator_deallocate_real8_5d
    module procedure :: gator_deallocate_real8_6d
    module procedure :: gator_deallocate_real8_7d
    ...
end interface
```
This will throw a compile-time error if `real4` is promoted to `real8`:
```
/ascldap/users/bhillma/codes/scream/branches/enable-scream-rrtmgp/externals/YAKL/gator_mod.F90:449:36:

  449 |   subroutine gator_allocate_real4_7d( arr , dims , lbounds_in )
      |                                    1
......
  562 |   subroutine gator_allocate_real8_7d( arr , dims , lbounds_in )
      |                                    2
Error: Ambiguous interfaces in generic interface 'gator_allocate' for ‘gator_allocate_real4_7d’ at (1) and ‘gator_allocate_real8_7d’ at (2)
/ascldap/users/bhillma/codes/scream/branches/enable-scream-rrtmgp/externals/YAKL/gator_mod.F90:799:38:

  799 |   subroutine gator_deallocate_real4_7d( arr )
      |                                      1
......
  835 |   subroutine gator_deallocate_real8_7d( arr )
      |                                      2
Error: Ambiguous interfaces in generic interface 'gator_deallocate' for ‘gator_deallocate_real4_7d’ at (1) and ‘gator_deallocate_real8_7d’ at (2)
```